src/shred_network/duplicate_shred_listener.zig

```diff
@@ -0,0 +1,794 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const Allocator = std.mem.Allocator;
+const Atomic = std.atomic.Value;
+const Channel = sig.sync.Channel;
+const GossipTable = sig.gossip.GossipTable;
+const Logger = sig.trace.Logger("duplicate_shred_listener");
+const Pubkey = sig.core.Pubkey;
+const RwMux = sig.sync.RwMux;
+const Slot = sig.core.Slot;
+
+const GossipVersionedData = sig.gossip.data.GossipVersionedData;
+const DuplicateShredData = sig.gossip.data.DuplicateShred;
+
+const ResultWriter = sig.ledger.ResultWriter;
+const LedgerReader = sig.ledger.Reader;
+const DuplicateSlotProof = sig.ledger.meta.DuplicateSlotProof;
+
+pub const GOSSIP_SLEEP_MILLIS: u64 = 100;
+pub const MAX_NUM_CHUNKS: usize = 3;
+pub const MAX_NUM_ENTRIES_PER_PUBKEY: usize = 128;
+pub const BUFFER_CAPACITY: usize = 512 * MAX_NUM_ENTRIES_PER_PUBKEY;
+
+pub const Params = struct {
+    exit: *Atomic(bool),
+    gossip_table_rw: *RwMux(GossipTable),
+    result_writer: ResultWriter,
+    ledger_reader: LedgerReader,
+    duplicate_slots_sender: *Channel(Slot),
+    leader_schedule: sig.core.leader_schedule.SlotLeaders,
+    shred_version: *const std.atomic.Value(u16),
+    epoch_schedule: sig.core.EpochSchedule,
+    epoch_ctx_mgr: *sig.adapter.EpochContextManager,
+};
+
+pub const DuplicateShredListener = struct {
+    thread: std.Thread,
+
+    pub fn start(
+        allocator: Allocator,
+        logger: Logger,
+        params: Params,
+    ) !DuplicateShredListener {
+        const thread = try std.Thread.spawn(
+            .{},
+            recvLoop,
+            .{ allocator, logger, params },
+        );
+        errdefer thread.join();
+
+        thread.setName("sigCiEntryLstnr") catch {};
+        return .{ .thread = thread };
+    }
+
+    pub fn join(self: DuplicateShredListener) void {
+        self.thread.join();
+    }
+};
```

Comment:
Is there any reason for this to exist? It doesn't appear to be used anywhere, everything just uses `recvLoop`.

Reply:
It was a remnant of a previous implemented. Forgot to clean it up after refactor. Removed now [eb9ca4a](https://github.com/Syndica/sig/pull/1027/commits/eb9ca4a660a2e0b3a3bbd007ea705a26d86d3b64)